src/__tests__/get-by-errors.js

```diff
@@ -39,6 +46,21 @@ cases(
   },
 )
 
+const getQueries = Object.keys(queries).filter(q => q.startsWith('get'))
+test.each(
+  getQueries.reduce((acc, query) => {
+    acc.push([query, true], [query, false])
+    return acc
+  }, []),
+)(
+  '%s query will throw expected error message when config.showLogOnFail is %s',
+  (query, showLogOnFail) => {
+    configure({showLogOnFail})
+    document.body.innerHTML = '<div>Hello</div>'
+    expect(() => screen[query]('TEST QUERY')).toThrowErrorMatchingSnapshot()
```

Comment:
I would prefer tests that specifically look for the presence/absence of the error message (not a fan of snapshot tests personally) but I'm not gonna block for it.  I'll defer to @kentcdodds  or someone else on that.

Reply:
Something like this? 
```javascript
const div = '<div>Hello</div>'
document.body.innerHTML = div
try {
    screen[query]('TEST QUERY'))
} catch (err) {
    expect(err.includes(div)).toBe(showLogOnFail)
}
```